Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -308,132 +415,342 @@ bool PPCSymbolDB::LoadMap(const std::string& filename, bool bad)
+    }
+
+    // Q&A: What is an Entry Symbol?
+    // In the following C Function featuring inline ASM, "__OSSystemCallVectorStart" and
+    // "__OSSystemCallVectorEnd" are Entry Symbols of the parent symbol, "SystemCallVector".
+
+    // void SystemCallVector()
+    // {
+    //   __asm {
+    //     __OSSystemCallVectorStart:
+    //     mfspr   r9, HID0
+    //     ori     r10, r9, 0x0008
+    //     mtspr   HID0, r10
+    //     isync
+    //     sync
+    //     mtspr   HID0, r9
+    //     rfi
+    //     __OSSystemCallVectorEnd:
+    //   }
+    // }
+
+    // The following is an abridged example of how Entry Symbols are recorded in a symbol map:
+    // 001f63c8 000020 801fb928  4 SystemCallVector
+    // 001f63c8 000000 801fb928 __OSSystemCallVectorStart (entry of SystemCallVector)
+    // 001f63e4 000000 801fb944 __OSSystemCallVectorEnd (entry of SystemCallVector)
+
+    // Notice that Entry Symbols have a reported size of 0 bytes, and show up in the middle of or
+    // at the start of other symbols.  Until Dolphin's Symbols / Symbol Database is improved to
+    // account for Entry Symbols properly, they are useless and even harmful to code highlighting.
+
+    // Also, compiler generated(?) Entry Symbols named things like "...data.0 (entry of .data)"
+    // are commonly children to Closure Symbols, which makes them doubly useless since Closure
+    // Symbols are being discarded atm.
+    if (is_entry)
+    {
+      ++entry_symbol_count;
+      continue;
+    }
+
+    // Function symbols final checks
+    if (section_name == ".init" || section_name == ".text")
+    {
+      type = Common::Symbol::Type::Function;
+
+      // Symbols outside of RAM are not valid entries.
+      // This also doubles as a check that the function's size is a multiple of 4
+      if (!PowerPC::HostIsInstructionRAMAddress(vaddress) ||
+          !(PowerPC::HostIsInstructionRAMAddress(vaddress + size - 4) && size != 0))
       {
-        // check for BLR before function
-        PowerPC::TryReadInstResult read_result = PowerPC::TryReadInstruction(vaddress - 4);
-        if (read_result.valid && read_result.hex == 0x4e800020)
+        ++invalid_address_count;
+        continue;
+      }
+
+      // Additional checks if the symbol map is suspected to be non-matching.
+      if (bad_map)
+      {
+        u32 prelude = PowerPC::HostRead_Instruction(vaddress - 4);
+        u32 ending = PowerPC::HostRead_Instruction(vaddress + size - 4);
+
+        // Things that can prelude a function symbol:
+        //  - BLR from previous function
+        //  - RFI from previous function
+        //  - Normal branch (TODO: to outside of function)
+        //     - Default function parameter signatures or handwritten ASM.
+        //  - NOP at end of function (inline ASM with naked end label)
+        //  - BCTR (__ptmf_scall)
+        //  - BLA (__OSDBJump)
+        //  - Padding null bytes (alignment)
+        if ((prelude != 0x4e800020) && (prelude != 0x4c000064) &&
```

Comment:
These shouldn't be hardcoded and can probably be merged with the other check bellow using a function.

Reply:
Done.  It is now being done with lambdas.